src/vivarium/framework/results/manager.py

```diff
@@ -1,21 +1,160 @@
-from typing import Callable, List
+from collections import Counter
+from typing import Callable, List, Union
 
 import pandas as pd
 
+# TODO: importing Builder causes circular import, fix that in implementation
+# from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.results.context import ResultsContext
+
 
 class ResultsManager:
+    """Backend manager object for the results management system.
+
+    The :class:`ResultManager` actually performs the actions needed to
+    stratify and observe results. It contains the public methods used by the
+    :class:`ResultsInterface` to register stratifications and observations,
+    which provide it with lists of methods to apply in their respective areas.
+    This Manager takes the place of the ResultsStratifier and all the
```

Comment:
No need to refer to previous functionality here. Focus on what this component actually does, which is provide methods to register stratifications and observations, and actually perform observations.